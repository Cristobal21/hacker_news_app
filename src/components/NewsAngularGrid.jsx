import { useEffect, useState } from "react"
import { useFetchDataAngular } from "../hooks/useFetchDataAngular"
import { NewsAngularItem } from "./NewsAngularItem"


export const NewsAngularGrid = () => {

    const [newsFilterA, setNewsFilterA] = useState([])
    const { newsAngular } = useFetchDataAngular()

    useEffect(() => {
        localStorage.setItem('newsA', JSON.stringify(newsAngular))
        const news = JSON.parse(localStorage.getItem('newsA'))

        if ( news ) {
            setNewsFilterA(news)
        }
    }, [newsAngular])
    
    return (
        <div className="container-box-news">
            {
                newsFilterA.map(data => (
                    <NewsAngularItem
                        key={data.id}
                        {...data}
                    />
                ))
            }
        </div>
    )
}
