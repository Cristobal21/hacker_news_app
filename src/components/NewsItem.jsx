import PropTypes from 'prop-types'
import { BiTimeFive } from "react-icons/bi"

export const NewsItem = ({ author, created, title, url }) => {

    const newsDate = created.slice(0,10)
    const newsFullHour = created.slice(11,19)
    const newsHourMin = newsFullHour.slice(0,5)

    const handleClick = () => {
        $(document).ready(() => {
            $("img").click((e) => {
                $("img").toggleClass("backImage")
            })
        })
    }

    const handleClickContainerText = () => {
        window.open(url,"_blank")
    }

    {
        if ( author, created, title, url ) {
            return (
                <>
                    <div className="container-news" >
                        <div className="container-text" onClick={ handleClickContainerText }>
                            <p className="p1"><BiTimeFive/>{` Posted on ${newsDate} at ${newsHourMin} by ${author}`}</p>
                            <p className="p2">{title}</p>
                        </div>
                        <div className="container-liked-button">
                            <div className="liked-button" id="containerHeartId">
                                <img 
                                    src="../../public/imgs/outline-heart.svg" 
                                    alt="heart" 
                                    className="img-liked"
                                    id="heartId"
                                    onClick={handleClick}
                                />
                            </div>
                        </div>
                    </div>
                </>
            )
        }
    }
}


NewsItem.propTypes = {
    author: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
