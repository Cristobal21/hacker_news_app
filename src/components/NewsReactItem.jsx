
export const NewsReactItem = ({ author, created, title, url }) => {


    return (
        <>
            <div className="container-news">
                <div className="container-text">
                    <p>{`${created} ago by ${author}`}</p>
                    <p>{title}</p>
                </div>
                <div className="container-liked-button">
                    <div className="liked-button"></div>
                </div>
            </div>
        </>
    )
}
