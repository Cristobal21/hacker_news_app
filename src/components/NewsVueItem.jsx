
export const NewsVueItem = ({ author, created, title, url }) => {

    const handleClick = () => {
        $(document).ready(() => {
            $("img").click((e) => {
                $("img").toggleClass("backImage")
            })
        })
    }

    const handleClickContainerText = () => {
        window.open(url,"_blank")
    }
    

    {
        if ( author, created, title, url ) {
            return (
                <>
                    <div className="container-news" >
                        <div className="container-text" onClick={ handleClickContainerText }>
                            <p>{`${created} ago by ${author}`}</p>
                            <p>{title}</p>
                        </div>
                        <div className="container-liked-button">
                            <div className="liked-button" id="containerHeartId">
                                <img 
                                    src="../public/imgs/outline-heart.svg" 
                                    alt="heart" 
                                    className="img-liked"
                                    id="heartId"
                                    onClick={handleClick}
                                />
                            </div>
                        </div>
                    </div>
                </>
            )
        }
    }
}
