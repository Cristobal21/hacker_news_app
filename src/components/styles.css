/* Dropdown coponent styles */

.container-dropdown {
    margin-bottom: 38px;
}

.dropdown-container {
    position: relative;
    display: inline-block;
    margin: 63px 0px 0px 0px;
}

.dropbtn {
    width: 240px;
    height: 32px;
    margin: 0;
    padding: 5px 12px 5px 12px;
    border-radius: 4px;
    background-color: #fff;
    font-family: 'Roboto', Roboto-Regular, sans-serif;
    font-size: 14px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.57;
    letter-spacing: normal;
    color: #343434;
    cursor: pointer;
    text-align: left;
}

.dropdown-content {
    display: none;
    position: absolute;
    background-color: #ffffff;
    min-width: 160px;
    overflow: auto;
    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
    z-index: 1;
    margin: 0;
}

.dropdown-content a {
    color: #343434;
    font-size: 14px;
    font-family: 'Roboto', Roboto-Regular, sans-serif;
    padding: 6px 89px 10px 11px;
    text-decoration: none;
    display: flex;
    align-items: center;
    background-color: #ffffff;
    width: 139px;
}

.dropdown-content a p {
    margin-left: 13px;
}

.dropdown-content a:hover {
    background-color: #eaeaea
}

.icon-container {
    color: black;
    width: 14px;
    height: 14px;
    margin: -23px 0px 0px 190px;
    padding: 2px 4px 3px;
}

#icon-style {
    margin-left: 90px;
}

.show {
    display: block;
}



/* News Styles */

.container-box-news {
    margin-left: 150px;
    margin-right: 150px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 40px;
    justify-items: center;
}

.container-news {
    width: 100%;
    height: 90px;
    margin: 18px 0px 30px 0px;
    padding: 0 0 0 0px;
    opacity: 0.8;
    border-radius: 6px;
    border: solid 1px #979797;
    background-color: #fff;
    display: flex;
}

.container-news:hover {
    opacity: 0.52;
    cursor: pointer;
}

.container-text {
    max-width: 100%;
    width: 100%;
}

.container-text p {
    margin-left: 30px;
    margin-right: 10px;
}

.container-liked-button {
    display: flex;
    justify-content: end;
}

.liked-button {
    width: 68px;
    height: 88px;
    margin: 0;
    padding: 0;
    border-radius: 6px;
    border: solid 1px #d9d9d9;
    background-color: #d9d9d9;
    display: flex;
    justify-content: center;
    align-items: center;
}

.liked-button img:hover {
    cursor: pointer;
}

.p1 {
    width: 100%;
    margin: 1.125rem 19.313rem 0.438rem 0.5rem;
    font-family: Roboto;
    font-size: 0.688rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #767676;
}

.p2 {
    width: auto;
    height: auto;
    font-family: Roboto;
    font-size: 0.875rem;
    font-weight: 500;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.43;
    letter-spacing: 0.25px;
    color: var(--brownish-grey);
}



/* Heart Image */

.backImage {
    background-image: url("../../public/imgs/heart.svg");
}


/* Media Query */

@media only screen and (min-width: 768px) and (max-width: 3840px) {
    .dropdown-container {
        margin-left: 150px;
        margin-right: 150px;
    }
}

@media only screen and (max-width: 1200px) {
    .container-dropdown {
        display: flex;
        justify-content: center;
    }
}

@media only screen and (max-width: 1440px) {
    .container-grid-news {
        display: flex;
        justify-content: center;
        margin-left: 30px;
        margin-right: 30px;
    }

    .container-box-news {
        margin-left: 0px;
        margin-right: 0px;
        display: flex;
        flex-direction: column;
        max-width: 550px;
        width: 100%;
    }
}

@media only screen and (max-width: 768px) {
    .container-news {
        width: auto;
        height: auto;
    }

    .liked-button {
        height: auto;
    }

    .container-text p {
        margin-right: 20px;
        margin-left: 20px;
    }
}