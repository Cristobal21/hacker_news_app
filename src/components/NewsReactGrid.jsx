import { useEffect, useState } from "react"
import { useFetchDataReact } from "../hooks/useFetchDataReact"
import { NewsItem } from "./NewsItem"

export const NewsReactGrid = () => {

    const [newsFilterR, setNewsFilterR] = useState([])
    const { newsReact } = useFetchDataReact()


    useEffect(() => {
        localStorage.setItem('newsR', JSON.stringify(newsReact))
        const news = JSON.parse(localStorage.getItem('newsR'))

        if ( news ) {
            setNewsFilterR(news)
        }

    }, [newsReact])
    

    return (
        <div className="container-box-news">
            {
                newsFilterR.map(data => (
                    <NewsItem
                        key={data.id}
                        {...data}
                    />
                ))
            }
        </div>
    )

}
