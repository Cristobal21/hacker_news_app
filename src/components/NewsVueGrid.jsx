import { useEffect, useState } from "react"
import { useFetchDataVuejs } from "../hooks/useFetchDataVuejs"
import { NewsVueItem } from "./NewsVueItem"

export const NewsVueGrid = () => {

    const [newsFilterV, setNewsFilterV] = useState([])
    const { newsVuejs } = useFetchDataVuejs()

    useEffect(() => {
        localStorage.setItem('newsV', JSON.stringify(newsVuejs))
        const news = JSON.parse(localStorage.getItem('newsV'))

        if ( news ) {
            setNewsFilterV(news)
        }
    }, [newsVuejs])
    
    
    
    
    return (
        <div className="container-box-news">
            {
                newsFilterV.map( data => (
                    <NewsVueItem
                        key={data.id}
                        { ...data }
                    />
                ))
            }
        </div>
    )
    
}
