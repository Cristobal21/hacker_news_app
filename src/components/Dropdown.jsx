import { Link } from 'react-router-dom'
import './styles.css'



export const Dropdown = () => {

    const handleClick = () => {
        document.getElementById("idDropdownContent").classList.toggle("show");
    }

    window.onclick = function (event) {
        if (!event.target.matches('.dropbtn')) {
            var dropdowns = document.getElementsByClassName("dropdown-content");
            var i;
            for (i = 0; i < dropdowns.length; i++) {
                var openDropdown = dropdowns[i];
                if (openDropdown.classList.contains('show')) {
                    openDropdown.classList.remove('show');
                }
            }
        }
    }

    return (
        <>
        <div className='container-dropdown'>
            <div className="dropdown-container">
                <button className="dropbtn" onClick={handleClick}>
                    Select your news
                    <i id='icon-style' className="fa-solid fa-angle-down"></i>
                </button>
                <div id='idDropdownContent' className='dropdown-content'>
                    <Link to="/angular">
                        <img src="../../public/imgs/angular.jpg" alt="angular" />
                        <p>Angular</p>
                    </Link>
                    <Link to="/react">
                        <img src="../../public/imgs/react.jpg" alt="react" />
                        <p>React</p>
                    </Link>
                    <Link to="/vuejs">
                        <img src="../../public/imgs/vuejs.jpg" alt="vuejs" />
                        <p>Vuejs</p>
                    </Link>
                </div>
            </div>
        </div>
        </>
    )
}
