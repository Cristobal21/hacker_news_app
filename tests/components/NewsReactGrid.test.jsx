import { render } from "@testing-library/react"
import { NewsReactGrid } from "../../src/components/NewsReactGrid"

describe('Test <NewsReactGrid/>', () => { 

    test('Match with snapshot', () => { 
        const { container } = render(<NewsReactGrid/>)
        expect( container ).toMatchSnapshot()
    })
})