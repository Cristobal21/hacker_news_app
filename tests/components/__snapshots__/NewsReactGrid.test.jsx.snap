// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <NewsReactGrid/> Match with snapshot 1`] = `
<div>
  <div
    class="container-box-news"
  />
</div>
`;
